tests/DCPS/DomainRange/DomainRangeTest.cpp

```diff
@@ -172,220 +143,207 @@ ACE_TMAIN(int argc, ACE_TCHAR *argv[])
+                          ACE_TEXT(" create_datawriter failed!\n")),
+                        -1);
+      }
 
-    // Create DataReaders
-    DDS::DataReaderListener_var listener(new DataReaderListenerImpl(reader_id, n_msgs, reader_done_callback));
+      // Create DataReaders
+      DDS::DataReaderListener_var listener(new DataReaderListenerImpl(reader_id, MSGS_PER_WRITER, reader_done_callback));
 
-    DDS::DataReaderQos dr_qos;
-    subscriber->get_default_datareader_qos(dr_qos);
-    dr_qos.reliability.kind = reliable ? DDS::RELIABLE_RELIABILITY_QOS : DDS::BEST_EFFORT_RELIABILITY_QOS;
+      DDS::DataReaderQos dr_qos;
+      subscriber->get_default_datareader_qos(dr_qos);
+      dr_qos.reliability.kind = DDS::RELIABLE_RELIABILITY_QOS;
 
-    DDS::DataReader_var reader =
-      subscriber->create_datareader(topic,
-                                    dr_qos,
-                                    listener,
-                                    OpenDDS::DCPS::DEFAULT_STATUS_MASK);
+      DDS::DataReader_var reader =
+        subscriber->create_datareader(topic,
+                                      dr_qos,
+                                      listener,
+                                      OpenDDS::DCPS::DEFAULT_STATUS_MASK);
 
-    if (!reader) {
-      ACE_ERROR_RETURN((LM_ERROR,
-                        ACE_TEXT("ERROR: %N:%l: main() -")
-                        ACE_TEXT(" create_datareader failed!\n")), -1);
-    }
+      if (!reader) {
+        ACE_ERROR_RETURN((LM_ERROR,
+                          ACE_TEXT("ERROR: %N:%l: main() -")
+                          ACE_TEXT(" create_datareader failed!\n")), -1);
+      }
 
-    Messenger::MessageDataReader_var reader_i =
-      Messenger::MessageDataReader::_narrow(reader);
+      Messenger::MessageDataReader_var reader_i =
+        Messenger::MessageDataReader::_narrow(reader);
 
-    if (!reader_i) {
-      ACE_ERROR_RETURN((LM_ERROR,
-                        ACE_TEXT("ERROR: %N:%l: main() -")
-                        ACE_TEXT(" _narrow failed!\n")),
-                       -1);
-    }
+      if (!reader_i) {
+        ACE_ERROR_RETURN((LM_ERROR,
+                          ACE_TEXT("ERROR: %N:%l: main() -")
+                          ACE_TEXT(" _narrow failed!\n")),
+                        -1);
+      }
 
-    ACE_DEBUG((LM_DEBUG, "(%P|%t) Spawning writer task\n", writer_id.c_str()));
-    WriterTask task(writer_id, writer, TOTAL_READERS);
-    task.activate(DEFAULT_FLAGS, TOTAL_WRITERS);
-    task.wait();
+      ACE_DEBUG((LM_DEBUG, "(%P|%t) Spawning writer task\n", writer_id.c_str()));
+      WriterTask task(writer_id, writer, TOTAL_READERS);
+      task.activate(DEFAULT_FLAGS, TOTAL_WRITERS);
+      task.wait();
 
-    if (!reliable)
-      ACE_OS::sleep(10);
-    else {
       ACE_Guard<ACE_Thread_Mutex> g(readers_done_lock);
       while (readers_done != TOTAL_READERS)
         readers_done_cond.wait();
+
       // Sleep allows an ACKNACK to be generated.
       ACE_OS::sleep(3);
```

Comment:
Rather than just sleep, can you either wait_for_acknowledgments() on the writer or wait for publication_match_status or subscription_match_status to go to 0?